src/request-helpers/hosted-request-helper.js

```diff
@@ -1,40 +1,54 @@
-import {responseToJSON} from "../utils/fetch/response-to-json";
+import {
+  fetchAuthorizationCode,
+  timeRemainingBeforeAccessTokenExpiresInSeconds,
+} from "../jira-oidc-helpers/auth";
+import { responseToJSON } from "../utils/fetch/response-to-json";
 
 function fetchFromLocalStorage(key) {
   return window.localStorage.getItem(key);
-};
+}
 
 async function fetchJSON(url, options) {
-	return fetch(url, options).then(responseToJSON)
+  return fetch(url, options).then(responseToJSON);
 }
 
-export function getHostedRequestHelper({ JIRA_API_URL }) {
-  return function(urlFragment) {
-    return new Promise(async(resolve, reject) => {
+export function getHostedRequestHelper(config) {
+  const { JIRA_API_URL } = config;
+
+  return function (urlFragment) {
+    return new Promise(async (resolve, reject) => {
       try {
-        const scopeIdForJira = fetchFromLocalStorage('scopeId');
-        const accessToken = fetchFromLocalStorage('accessToken');
-  
+        const scopeIdForJira = fetchFromLocalStorage("scopeId");
+        const accessToken = fetchFromLocalStorage("accessToken");
+
+        if (accessToken) {
+          const timeLeft = timeRemainingBeforeAccessTokenExpiresInSeconds();
+          const FIVE_SECONDS = 5;
+
+          if (timeLeft < FIVE_SECONDS) {
+            alert(
+              "Your access token needs to be refreshed. Taking you to Atlassian to reauthorize"
+            );
+            fetchAuthorizationCode(config);
+            return;
```

Comment:
Will returning `undefined` instead of a `Promise that never resolves` cause exceptions and possibly trigger sentry?

Reply:
I don't think so, but i'll add a promise so it doesn't